src/protocols/shell.rs

```diff
@@ -0,0 +1,292 @@
+//! Shell Protocol
+//!
+//! Provides shell services to UEFI applications.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x6302d008,
+    0x7f9b,
+    0x4f30,
+    0x87,
+    0xac,
+    &[0x60, 0xc9, 0xfe, 0xf5, 0xda, 0x4e],
+);
+
+pub type ShellFileHandle = *mut core::ffi::c_void;
+
+pub type DeviceNameFlags = u32;
+pub const USE_COMPONENT_NAME: DeviceNameFlags = 0x00000001;
+pub const USE_DEVICE_PATH: DeviceNameFlags = 0x00000002;
+
+#[repr(C)]
+pub struct ListEntry {
+    pub flink: *mut ListEntry,
+    pub blink: *mut ListEntry,
+}
+
+#[repr(C)]
+pub struct ShellFileInfo {
```

Comment:
This is called `EFI_SHELL_FILE_INFO` in the spec, so we should really name it `FileInfo`.